url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/io/TestForByteArrayBuilder.java

```diff
@@ -0,0 +1,191 @@
+	public static int LEVEL = 30;
+
+	private void log(String log) {
+//		System.out.println("#TestForByteArrayBuilder#"+log);
+	}
+
+
+	public void testHello() {
+		log("testHello");
+	}
+
+	public void testParsrShort() {
+		byte[] portAsBytes = ByteArrayBuilder.parseShort(8080, ByteArrayBuilder.BYTEORDER_BIG_ENDIAN);		
+		int portAsShort = ByteArrayBuilder.parseShort(portAsBytes, 0, ByteArrayBuilder.BYTEORDER_BIG_ENDIAN);
+		System.out.println(""+portAsBytes[0]+","+portAsBytes[1]);
+		System.out.println(""+portAsShort);
+		ByteBuffer buffer = ByteBuffer.allocate(2);
+		buffer.order(ByteOrder.BIG_ENDIAN);
+		buffer.putShort((short)8080);
+		buffer.flip();
+		assertEquals(ByteArrayBuilder.parseShort(buffer.array(), 0, ByteArrayBuilder.BYTEORDER_BIG_ENDIAN), portAsShort);
+	}
+
+	public void testAppend() {
+		ByteArrayBuilder builder = new ByteArrayBuilder();
+		byte[] testdata = new byte[1024];
+		for(int i=0;i<testdata.length;i++) {
+			testdata[i] = (byte)i;
+		}
+
+		for(int i=0;i<testdata.length;i++) {
+			assertEquals(i, builder.length());
+			builder.append(testdata[i]);
+			// check
+			{
+				byte[] result = builder.getBuffer(); 
+				int s = 0;
+				int e = builder.length();
+				assertEquals(i+1, e);
+				if(i==0||i%LEVEL==0){
+					for(int j=s;j<e;j++) {
+						assertEquals((byte)j, result[j]);
+					}
+				}
+			}
+		}		
+	}
+
+	public void testClear() {
+		ByteArrayBuilder builder = new ByteArrayBuilder();
+		byte[] testdata = new byte[1024];
+		for(int i=0;i<testdata.length;i++) {
+			testdata[i] = (byte)i;
+		}
+
+		builder.clear();
+		for(int i=0;i<testdata.length;i++) {
+			assertEquals(i, builder.length());
+			builder.append(testdata[i]);
+			// check
+			{
+				byte[] result = builder.getBuffer(); 
+				int s = 0;
+				int e = builder.length();
+				assertEquals(i+1, e);
+				if(i==0||i%LEVEL==0){
+					for(int j=s;j<e;j++) {
+						assertEquals((byte)j, result[j]);
+					}
+				}
+			}
+		}		
+		builder.clear();
+		for(int i=0;i<testdata.length;i++) {
+			assertEquals(i, builder.length());
+			builder.append(testdata[i]);
+			// check
+			{
+				byte[] result = builder.getBuffer(); 
+				int s = 0;
+				int e = builder.length();
+				assertEquals(i+1, e);
+				if(i==0||i%LEVEL==0){
+					for(int j=s;j<e;j++) {
+						assertEquals((byte)j, result[j]);
+					}
+				}
+			}
+		}
+	}
+
+	public void testSetBufferLength() {
+		ByteArrayBuilder builder = new ByteArrayBuilder();
+		byte[] testdata = new byte[777];
+		for(int i=0;i<testdata.length;i++) {
+			testdata[i] = (byte)i;
+		}
+
+		builder.clear();
+		for(int i=0;i<testdata.length;i++) {
+			assertEquals(i, builder.length());
+			builder.append(testdata[i]);
+			builder.setBufferLength(i+2);
+			// check
+			{
+				byte[] result = builder.getBuffer(); 
+				int s = 0;
+				int e = builder.length();
+				assertEquals(i+1, e);
+				if(i==0||i%LEVEL==0){
+					for(int j=s;j<e;j++) {
+						assertEquals((byte)j, result[j]);
+					}
+				}
+			}
+		}		
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 113:129, 131:147.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.